### ISSUE TITLE
Matplotlib warning about using FixedFormatter with FixedLocator

### ISSUE THREAD

**Issue by aanil** [ewels/MultiQC#1323]
**Description of bug:**

Multiqc 1.9 when run with the ngi template on irma displays the warning `UserWarning: FixedFormatter should only be used together with FixedLocator` multiple times. However the multiqc report is still generated. Could be related to matplotlib.

**MultiQC Error log:**

```
/sw/anaconda/envs/py3/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
 axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])
/sw/anaconda/envs/py3/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
 axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])
/sw/anaconda/envs/py3/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
 axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])
/sw/anaconda/envs/py3/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
 axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])
/sw/anaconda/envs/NGI_py3/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
 axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])
[INFO  ]    bcftools : Found 16 stats reports
/sw/anaconda/envs/py3/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
 axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])
[INFO  ]     picard : Found 8 MarkDuplicates reports
/sw/anaconda/envs/py3/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
 axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])

```

**File that triggers the error:**

**MultiQC run details (please complete the following):**
* Command used to run MultiQC:  multiqc -c multiqc_sthlm_config.yml qc_ngi
* MultiQC Version:  1.9
* Operating System:  Linux (Irma's current)
* Python Version:  Python 3.8.5
* Method of MultiQC installation:  pip

**Additional context**

**Comment by guidohooiveld** [comment]
Although I am using the dev version of MultiQC (1.10.dev0), I saw the same warning. To me it seems that this impacts 'playing' with the the table layout in the HTML page.,..

Did some googling, and found this thread [here](https://stackoverflow.com/questions/63723514/userwarning-fixedformatter-should-only-be-used-together-with-fixedlocator), which indicated that this warning is due to `matplotlib`. I followed its suggestion (i,e. downgraded to matplotlib v3.2.2), and the warning was gone...

Thus: this gives a warning:
```
[guidoh@localhost Mira]$  multiqc ./data_out --filename multiqc_report.html --title "Experiment A415" --export
[INFO   ]         multiqc : This is MultiQC v1.10.dev0
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Report title: Experiment A415
[INFO   ]         multiqc : Searching   : /mnt/files/guido/Mira/data_out
Searching 2478 files..  [####################################]  100%          
[INFO   ]          salmon : Found 42 meta reports
[INFO   ]          salmon : Found 42 fragment length distributions
[INFO   ]           fastp : Found 42 reports
/home/guidoh/.local/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
  axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])
[INFO   ]          fastqc : Found 84 reports
/home/guidoh/.local/lib/python3.8/site-packages/multiqc/plots/bargraph.py:451: UserWarning: FixedFormatter should only be used together with FixedLocator
  axes.set_xticklabels(['{:.0f}%'.format(x) for x in vals])
[INFO   ]         multiqc : Compressing plot data
[INFO   ]         multiqc : Report      : multiqc_report.html
[INFO   ]         multiqc : Data        : multiqc_report_data
[INFO   ]         multiqc : Plots       : multiqc_report_plots
[INFO   ]         multiqc : MultiQC complete
[guidoh@localhost Mira]$
```

Downgraded matplotlib to v3.**2**.2 (from 3.**3**.2)
```
[guidoh@localhost Mira]$ python -m pip install matplotlib==3.2.2 --user
WARNING: Keyring is skipped due to an exception: Failed to unlock the collection!
Collecting matplotlib==3.2.2
  Using cached https://files.pythonhosted.org/packages/77/5c/6a3aa574ed33911a0b4deb66b3b9d2e72ed4bcbfd70f52a5546395156dd3/matplotlib-3.2.2-cp38-cp38-manylinux1_x86_64.whl
Requirement already satisfied: cycler>=0.10 in /home/guidoh/.local/lib/python3.8/site-packages (from matplotlib==3.2.2) (0.10.0)
Requirement already satisfied: python-dateutil>=2.1 in /home/guidoh/.local/lib/python3.8/site-packages (from matplotlib==3.2.2) (2.8.1)
Requirement already satisfied: kiwisolver>=1.0.1 in /home/guidoh/.local/lib/python3.8/site-packages (from matplotlib==3.2.2) (1.3.1)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /home/guidoh/.local/lib/python3.8/site-packages (from matplotlib==3.2.2) (2.4.7)
Requirement already satisfied: numpy>=1.11 in /home/guidoh/.local/lib/python3.8/site-packages (from matplotlib==3.2.2) (1.19.4)
Requirement already satisfied: six in /home/guidoh/.local/lib/python3.8/site-packages (from cycler>=0.10->matplotlib==3.2.2) (1.15.0)
Installing collected packages: matplotlib
  Found existing installation: matplotlib 3.3.2
    Uninstalling matplotlib-3.3.2:
      Successfully uninstalled matplotlib-3.3.2
Successfully installed matplotlib-3.2.2
```
 
Then reran MultiQC without warnings/ problems!
```
[guidoh@localhost Mira]$  multiqc ./data_out --filename multiqc_report.html --title "Experiment A415" --export
[INFO   ]         multiqc : This is MultiQC v1.10.dev0
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Report title: Experiment A415
[INFO   ]         multiqc : Searching   : /mnt/files/guido/Mira/data_out
Searching 2478 files..  [####################################]  100%          
[INFO   ]          salmon : Found 42 meta reports
[INFO   ]          salmon : Found 42 fragment length distributions
[INFO   ]           fastp : Found 42 reports
[INFO   ]          fastqc : Found 84 reports
[INFO   ]         multiqc : Compressing plot data
[INFO   ]         multiqc : Report      : multiqc_report.html
[INFO   ]         multiqc : Data        : multiqc_report_data
[INFO   ]         multiqc : Plots       : multiqc_report_plots
[INFO   ]         multiqc : MultiQC complete
[guidoh@localhost Mira]$

```

MultiQC run details (please complete the following):

- Command used to run MultiQC: `multiqc ./data_out --filename multiqc_report.html --title "Experiment A415" --export`
- MultiQC Version: `1.10.dev0`
- Operating System: `Fedora release 32 (Thirty Two)`
- Python Version: `Python 3.8.6`
- Method of MultiQC installation:  `pip install --user --upgrade --force-reinstall git+https://github.com/ewels/MultiQC.git`

**Comment by ewels** [comment]
Fixed in the v1.10 release in commit https://github.com/ewels/MultiQC/commit/de0f01b5db483c64ebba45e26e3984743a5e26c5
